src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -358,5 +358,19 @@ public static void RegisterPackages (string[] packages, Converter<string, Type>[
 				}
 			}
 		}
+
+		public class JavaTypeManager : JavaObject
```

Comment:
Can this type be `internal` instead of `public`?

Reply:
Hopefully yes. I think it should be only used from Mono.Android and the generated code, which should end up in the Mono.Android as well.

I will test whether it will indeed work in the real apk.

Reply:
It better work, or we'd implicitly be requiring that *all* `Java.Lang.Object` subclasses in all assemblies be `public`, which is *not* a pre-existing requirement. :-)

Reply:
tested and works OK here. with new marshal methods and in fallback as well.